### PR TITLE
sqlc: adopt

### PR DESCRIPTION
**Pull request by aaronjheng**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by zi3m5f — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349728`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sqlc</li>
  </ul>
</details>

File: pkgs/by-name/sq/sqlc/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      --bash <($out/bin/sqlc completion bash) \
+      --fish <($out/bin/sqlc completion fish) \
+      --zsh <($out/bin/sqlc completion zsh)
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = sqlc;
+    command = "sqlc version";
+    version = "v${version}";
+  };
```
[the remark this answers is not in the document]
Comment (aaronjheng) [reply pkgs/by-name/sq/sqlc/package.nix:48]: Yeah, let's keep it for now.

**Comment by aaronjheng** [comment]
> Splitting your changes into separate commits would be great too:
> 
> https://github.com/NixOS/nixpkgs/blob/86c19dd026cd371140b6a5b9f7e6188830ee2621/CONTRIBUTING.md?plain=1#L497
> 
> For example:
> 
> * sqlc: migrate to pkgs/by-name
> * sqlc: nixfmt-rfc-style formatting
> * .git-blame-ignore-revs: add "sqlc: nixfmt-rfc-style formatting"
> * sqlc: install shell completions
> * sqlc: "add testers.testVersion test" or "use versionCheckHook"

This PR is more like a rewrite. How about making the PR description more detailed?

**Comment by aaronjheng** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/351796

**Comment by aaronjheng** [comment]
@zi3m5f

I’m really sorry for how I handled that PR. I didn’t mean to overlook your time and effort—your input was valuable and made a real difference. I think that’s also why the new PR could be merged so quickly. Thanks for your understanding and for your contributions! :)

**Pull request closed without merging**